src/commands/domains/add.ts

```diff
@@ -0,0 +1,73 @@
+import { Command, flags } from '@oclif/command'
+
+import * as api from '../../api'
+import setupApiClient from '../../setupApiClient'
+
+import { handle } from 'oazapfts'
+import withStandardErrors from '../../utils/errorHandling'
+
+type FlagsType = {
+  domain?: string
+
+  [key: string]: any
+}
+
+export default class DomainsAdd extends Command {
+  static description = 'add a domain'
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    domain: flags.string({
+      char: 'd',
+      description: 'the domain to add to your account',
+    }),
+  }
+
+  static args = []
+
+  async run() {
+    const { flags } = this.parse(DomainsAdd)
+
+    const client = await setupApiClient()
+
+    if (!client) {
+      this.log('Could not connect to API. Check your internet connection')
+      this.exit(1)
+    }
+
+    return this.add(client, flags)
+  }
+
+  async add(client: typeof api, { domain }: FlagsType): Promise<void> {
+    if (!domain) {
+      this.log('Please provide a domain to be setup')
+
+      this.exit(1)
+    }
+
+    return handle(
+      client.addDomain({ domain } as api.AddDomainRequest),
+      withStandardErrors(
+        {
+          '200'({ id, record_name, record_value }: api.AddDomainResponse) {
+            this.log(
+              'Please add the following DNS record to your domain,',
+              id + ',',
+              'to verify your ownership',
+            )
+            this.log()
+            this.log('Type: TXT')
+            this.log('Name:', record_name)
+            this.log('Value:', record_value)
+            this.log()
+            this.log('Type: MX')
```

Comment:
What about DKIM and DMARC?

Reply:
Not sure, shouldn't that be provided by the api? Wait for me to finish current work on api before potential changes.

Reply:
DKIM and DMARC is assumed to be static. If they do change then we would have to notify the user of such changes a whole separate process. I think for now this isn't important... On another note we can provide a link to the docs https://docs.mailscript.com/cli/domains/